### ISSUE TITLE
Composer View misplaced everytime keyboard is shown

### ISSUE THREAD

**Issue by carlostrianag** [GetStream/stream-chat-swift#88]
## What did you do?
- Installed the SDK via Carthage (also tried with CocoaPods) into my project
- Opened the app
- Created a channel
- Tapped on the Composer View
- Once the keyboard is shown, there's a weird gap in between Composer View and keyboard

## What did you expect to happen?
Composer View should appear right on top of the Keyboard, no gaps should be displayed

## What happened instead?
Strange gap appeared in between keyboard and Composer View. See resources below.
https://pasteboard.co/ISJs66E.png
https://drive.google.com/file/d/1DAKaQ50DBbgwZCI_d4b08d2ztaD98p78/view?usp=sharing

## GetStream Environment
**GetStream Chat version:** 1.5.5
**iOS version:** 13.3
**Swift version:** 5.2.1
**Xcode version:** 11.2.1 (11B500)
**Device:** All devices

## Additional context
If you guys do not have the time for solving this soon, I would like to hear any insight you can provide on how could it be solved, and then I can try to do so, if it works I'd love to open the PR.

Any help would be much appreciated!

PS:  I have also read other issues such as https://github.com/GetStream/stream-chat-swift/issues/28 and https://github.com/GetStream/stream-chat-swift/issues/36 and it seems like it is still happening.

I can also say: my ChatViewController is embedded on a TabBarController if that helps.

**Comment by saurabhPV** [comment]
Gap is there even when keyboard is not visible.
<img width="444" alt="Screenshot 2020-02-03 at 11 30 41 AM" src="https://user-images.githubusercontent.com/15213067/73629315-b979ab80-4678-11ea-9290-b84738604c77.png">

**Comment by b-onc** [comment]
Hello @carlostrianag 
Thanks for the issue. Can you post more info about the view hierarchy? A screenshot of view debugger would help. Is your tabbar opaque? Do you directly embed chatviewcontroller or ChannelViewController pushes it?
We've resolved this issue in 1.5.5 as you've mentioned so it's surprising you're seeing this. If it's a regression, we'll make sure to resolve it in 1.5.6

@saurabhPV It's expected, you can customize this via `style`. See [here](https://github.com/GetStream/stream-chat-swift/wiki/Styles)

**Comment by carlostrianag** [comment]
Hello @b-onc, thanks for the quick response. Here are the details you requested.

### Can you post more info about the view hierarchy?
![image](https://user-images.githubusercontent.com/7474390/73667426-3d7c6500-4672-11ea-9278-d5ec0a289fb2.png)

### Is your tabbar opaque?
Yes, it is

### Do you directly embed chatviewcontroller or ChannelViewController pushes it?
ChannelViewController pushes it.

**Comment by b-onc** [comment]
@carlostrianag Thanks for the info. Can you share more about your `ChatDetailsViewController`? Is the tab bar visible in your `ChatDetailsViewController`?
If your code is proprietary you can email us on support@getstream.io

**Comment by carlostrianag** [comment]
@b-onc - `ChatDetailsViewController` is a a class that extends from `ChatViewController`.

I have one more class called `ConversationsViewController` which is a class that extends from your `ChannelsViewController` and there I have overridden the `createChatViewController` with this:
```
    override func createChatViewController(with channelPresenter: ChannelPresenter, indexPath: IndexPath) -> ChatViewController {
        let chatVC = ChatDetailsViewController()
        chatVC.style = style
        channelPresenter.eventsFilter = channelsPresenter.channelEventsFilter
        chatVC.channelPresenter = channelPresenter
        return chatVC
    }
```

### Is the tab bar visible in your ChatDetailsViewController?
No, tab bar is containing the `ChatDetailsViewController` through `ConversationsViewController` however it is not visible when `ChatDetailsViewController` is presented.


### View Hierarchy
_`UITabBarController` 
 ______ `NavigationController`
___________`ConversationsViewController`
_______________`ChatDetailsViewController`

**Comment by b-onc** [comment]
@carlostrianag Thanks! I suspect the issue lies somewhere in how the `ChatDetailsViewController` UI is setup - in our tests (which uses `ChatViewController` we can't reproduce this issue. What custom stuff are you doing (regarding UI) inside your `ChatDetailsViewController`?
And by the way thanks a lot for the informative and to the point comments 👍

**Comment by carlostrianag** [comment]
@b-onc - you're welcome!

### What custom stuff are you doing (regarding UI) inside your `ChatDetailsViewController`?

This is what I'm doing inside this class.

```
class ChatDetailsViewController: ChatViewController {
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        navigationController?.setupDefaultNavigationBar()
        navigationController?.navigationBar.prefersLargeTitles = false
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
    }
}
```

This is the code inside the method `setupDefaultNavigationBar()`
```
    func setupDefaultNavigationBar() {
        navigationBar.setBackgroundImage(UIColor.white.image(), for: .default)
        navigationBar.shadowImage = UIColor.white.image()
        navigationBar.isTranslucent = false
        navigationBar.tintColor = .black
        modalPresentationCapturesStatusBarAppearance = true
    }
```

**Comment by b-onc** [comment]
Hello @carlostrianag ,
Trying the exact same configuration you've provided, I'm unable to reproduce the issue: the `ComposerView` moves correctly. Can you try deleting `DerivedData` (and related cache) and clean install? It'd be awesome if you can provide a simple example project.

**Comment by carlostrianag** [comment]
Hi @b-onc - Deleted `DerivedData` even tested on a different computer, same issue. 

Here you will find an example project with the issue I'm describing, hope you are able to replicate the it using this!

https://drive.google.com/file/d/1zrc0a_JYdyKsTOpVTO4oi_VbO8OPdP2k/view?usp=sharing

### **IMPORTANT:**
Please, do not forget to fill file `SceneDelegate.swift` line `23` with your **API KEY.**

**Comment by buh** [comment]
Thanks @carlostrianag for the example project. Now we see the issue and will fix it in the next release.

**Comment by carlostrianag** [comment]
Hi @buh, glad to hear you were able to replicate it. I'd like to add: even if your next release isn't happening soon, I would like to have access to the commit where it is fix, to see if I can integrate in the project, only if this is possible for you guys.

Thanks in advance for your hard work and help with this!

**Comment by b-onc** [comment]
Hello @carlostrianag ,
We've investigated the issue, and found a solution: just add `.edgesIgnoringSafeArea(.bottom)`:
```
struct ContentView: View {
    var body: some View {
        TabView {
            ChatsViewController().tabItem {
                VStack {
                    Image(systemName: "1.circle")
                    Text("Chats")
                }
            }.edgesIgnoringSafeArea(.bottom)
        }
    }
}
```
Turns out SwiftUI does not pass safe area to child views, it just places them on top of tab bar and this was causing wrong calculations in composer view position calculations. When we add `edgesIgnoringSafeAre(.bottom)`, SwiftUI puts the view all the way to the bottom so safe area is included in the calculations.
You can check how it's calculated in `ChatViewController.swift:53` and `ChatViewController+Composer.swift:21`. It'd be helpful to mention you're using SwiftUI too, since we officially don't support it yet (but it's on our roadmap). Let me know if that works for you.

**Comment by carlostrianag** [comment]
Definitely will try that! I’ll let you know how it goes, and please excuse me for not telling you guys I was using SwiftUI.

Thanks!

**Comment by carlostrianag** [comment]
Hello @b-onc, it worked for SwiftUI, however I realized I should not be using SwiftUI for that view, so I tried the usual way, using Storyboards. What is weird is that if I try using Storyboards in the sample project I sent, it works perfectly, however if I try Storyboards in the actual project it won't work. Same thing keeps happening. Already cleaned and re-installed.

As this is proprietary code, is there a way we could have sort of a call, to find out what's wrong and determine if it is still a bug?
